R/initializeProcessInputs.R

```diff
@@ -18,30 +19,37 @@ defaultProcessInputs <- function(algorithm, wps_url, wps_version){
   checkException(doc)
   
   # -- get optional arguments --
-  optionNd	<-	getNodeSet(doc,'//DataInputs/Input[@minOccurs=0]/ows:Identifier')
+  optionNd	<-	xml2::xml_find_all(doc,'//DataInputs/Input[@minOccurs=0]/ows:Identifier')
   optionLs	<-	vector("list",length(optionNd))
   optionLs[]	<-	NA # "NA" is the equivalent of "optional" as an input
-  names(optionLs)	<-	sapply(optionNd,xmlValue)
+  names(optionLs)	<-	sapply(optionNd,xml2::xml_text)
   
   # -- get required arguments -- (minOccurs > 0 means required)
-  requirNd	<-	getNodeSet(doc,'//DataInputs/Input[@minOccurs>0]/ows:Identifier')
+  requirNd	<-	xml2::xml_find_all(doc,'//DataInputs/Input[@minOccurs>0]/ows:Identifier')
   requirLs	<-	vector("list",length(requirNd))	
-  names(requirLs)	<-	sapply(requirNd,xmlValue)
+  names(requirLs)	<-	sapply(requirNd,xml2::xml_text)
   
   # -- get and set default values --
   defaultTag <- '//DataInputs/Input/LiteralData/DefaultValue'
-  defaultNd	<-	getNodeSet(doc,paste0(defaultTag, '/parent::node()[1]/DefaultValue'))
-  defaultLs	<-	vector("list",length(sapply(defaultNd,xmlValue)))
-  defaultLs[]	<-	sapply(defaultNd,xmlValue)
-  names(defaultLs)	<-	sapply(getNodeSet(doc,paste0(defaultTag, '/parent::node()[1]/preceding-sibling::node()[3]')),xmlValue)	
+  defaultNd	<-	xml2::xml_find_all(doc,paste0(defaultTag, '/parent::node()[1]/DefaultValue'))
+  defaultLs	<-	vector("list",length(sapply(defaultNd,xml2::xml_text)))
+  defaultLs[]	<-	sapply(defaultNd,xml2::xml_text)
```

Comment:
same here, and a few others

Reply:
hmmm... ok I just left the sapply and things worked so I didn't worry about it. Will try and switch to straight xml_text call.

Reply:
Yeah, it probably still works fine with sapply

Reply:
I switch this and tests don't work... OK if I leave it?

Reply:
Hmm ok.  Yeah, fine for now at least.

Reply:
Actually, I take that back. I can eliminate a lot of the sapply calls. Will be in my latest testing PR.